File: convertedToIOS5renamed/ch10p218subclassUIView/p198p204subclassUIView/MyHorizLine.m


#import "MyHorizLine.h"


@implementation MyHorizLine

// this (initWithCoder) is new
// in the previous edition we were using a white rectangle on a white background, masking this issue
// but in the Single View Application template the background is grey

- (id)initWithCoder:(NSCoder *)decoder
{
    self = [super initWithCoder:decoder];
    if (self) {
        self.backgroundColor = [UIColor clearColor];
    }
    return self;
}

- (void)drawRect:(CGRect)rect
{
    CGContextRef c = UIGraphicsGetCurrentContext();
    CGContextMoveToPoint(c, 0, 0);
    CGContextAddLineToPoint(c, self.bounds.size.width, 0);
    CGContextStrokePath(c);
}


@end

File: bk1ch10p418subclassing/bk1ch10p418subclassing/MyHorizLine.swift


import UIKit

class MyHorizLine: UIView {

    required init?(coder aDecoder: NSCoder) {
        super.init(coder:aDecoder)
        self.backgroundColor = .clear
    }
    override func draw(_ rect: CGRect) {
        let c = UIGraphicsGetCurrentContext()!
        c.moveTo(x: 0, y: 0)
        c.addLineTo(x: self.bounds.size.width, y: 0)
        c.strokePath()
    }


}
